x/posts/keeper/common_functions.go

```diff
@@ -9,20 +9,42 @@ import (
 )
 
 // IteratePosts iterates through the posts set and performs the provided function
+// It makes a copy of the posts array which is done only for sorting purposes.
 func (k Keeper) IteratePosts(ctx sdk.Context, fn func(index int64, post types.Post) (stop bool)) {
 	store := ctx.KVStore(k.StoreKey)
 	iterator := sdk.KVStorePrefixIterator(store, types.PostStorePrefix)
 	defer iterator.Close()
-	i := int64(0)
+
+	var posts types.Posts
 	for ; iterator.Valid(); iterator.Next() {
 		var post types.Post
 		k.Cdc.MustUnmarshalBinaryBare(iterator.Value(), &post)
+		posts = append(posts, post)
+	}
+
+	i := int64(0)
+	postsSorted := make(types.Posts, len(posts))
+	for _, post := range posts {
+		var index sdk.Int
+		k.Cdc.MustUnmarshalBinaryBare(store.Get(types.PostIndexedIDStoreKey(post.PostID)), &index)
+		postsSorted[index.Int64()-1] = post
+	}
+
+	//freeing up memory
+	//nolint
+	posts = nil
+
+	for _, post := range postsSorted {
 		stop := fn(i, post)
 		if stop {
 			break
 		}
 		i++
 	}
+
+	//freeing up memory
+	//nolint
+	postsSorted = nil
```

Comment:
Since `postsSorted` is created inside its body, when the function returns it will be cleared up anyway. I suggest we remove this then

```suggestion
```